libtiledbsoma/src/utils/arrow_adapter.cc

```diff
@@ -435,6 +435,87 @@ std::unique_ptr<ArrowSchema> ArrowAdapter::arrow_schema_from_tiledb_array(
     return arrow_schema;
 }
 
+std::unique_ptr<ArrowSchema> ArrowAdapter::arrow_schema_from_tiledb_dimension(
+    const Dimension& dimension) {
+    std::unique_ptr<ArrowSchema> arrow_schema = std::make_unique<ArrowSchema>();
+    arrow_schema->format = strdup(
+        ArrowAdapter::to_arrow_format(dimension.type()).data());
+    arrow_schema->name = strdup(dimension.name().c_str());
+    arrow_schema->metadata = nullptr;
+    arrow_schema->flags = 0;
+    arrow_schema->n_children = 0;
+    arrow_schema->children = nullptr;
+    arrow_schema->dictionary = nullptr;
+    arrow_schema->release = &ArrowAdapter::release_schema;
+    arrow_schema->private_data = nullptr;
+    LOG_TRACE(std::format(
+        "[ArrowAdapter] arrow_schema_from_tiledb_dimension format {} "
+        "name {}",
+        arrow_schema->format,
+        arrow_schema->name));
+
+    return arrow_schema;
+}
+
+std::unique_ptr<ArrowSchema> ArrowAdapter::arrow_schema_from_tiledb_attribute(
+    Attribute& attribute, const Context& ctx, const Array& tiledb_array) {
+    std::unique_ptr<ArrowSchema> arrow_schema = std::make_unique<ArrowSchema>();
+    arrow_schema->format = strdup(
+        ArrowAdapter::to_arrow_format(attribute.type()).data());
+    arrow_schema->name = strdup(attribute.name().c_str());
+    arrow_schema->metadata = nullptr;
+    arrow_schema->flags = 0;
+    if (attribute.nullable() &&
+        strcmp(attribute.name().c_str(), SOMA_GEOMETRY_COLUMN_NAME.c_str()) !=
+            0) {
+        arrow_schema->flags |= ARROW_FLAG_NULLABLE;
+    } else {
+        arrow_schema->flags &= ~ARROW_FLAG_NULLABLE;
+    }
```

Comment:
```suggestion
    if (attribute.nullable() &&
        attribute.name() != SOMA_GEOMETRY_COLUMN_NAME) {
        arrow_schema->flags |= ARROW_FLAG_NULLABLE;
    } else {
        arrow_schema->flags &= ~ARROW_FLAG_NULLABLE;
    }
```
No need to convert both to C strings and `strcmp` when you can compare as C++ strings